packages/api-client/src/store/workspace.ts

```diff
@@ -287,23 +287,36 @@ export const createWorkspaceStore = (router: Router, persistData = true) => {
       },
     }
 
-    if (request.requestBody) {
-      const requestBody = getRequestBodyFromOperation({
+    const bodyParams = Object.values(request.parameters.body)
+
+    let requestBody
+
+    if (bodyParams.length) {
+      requestBody = getRequestBodyFromOperation({
+        httpVerb: request.method,
+        path: request.path,
+        information: {
+          parameters: bodyParams,
+        },
+      })
+    } else if (request.requestBody) {
+      requestBody = getRequestBodyFromOperation({
```

Comment:
@amritk This looks like it is redoing work from `getRequestBodyFromOperation` (or mutating it's logic?)

In that function we already check for body parameters and return it first. To get body parameters it uses the more complicated method of:
```ts
const bodyParameters = getParametersFromOperation(operation, 'body', false)
```
But otherwise does the same thing. 

If we truly want this logic (and not improving `getRequestBodyFromOperation`) then I would suggest:
```ts
const requestBody =
  bodyParams.length || request.requestBody
    ? getRequestBodyFromOperation({
        httpVerb: request.method,
        path: request.path,
        information: {
          requestBody: bodyParams.length ? bodyParams : request.requestBody,
        },
      })
    : undefined
```

However I think these changes need to be in the `getRequestBodyFromOperation`

Reply:
All of this logic needs significant commenting as well. The intent is a bit hard to follow here.